test/test_autograd.py

```diff
@@ -4499,6 +4502,105 @@ def fn(*inputs):
 
             setattr(TestAutogradDeviceType, test_name, do_test)
 
+class TestAutogradComplex(TestCase):
+    # remove this test after gradcheck support is added for non-holomoprphic functions
+    def test_real(self):
+        x = torch.randn(10, dtype=torch.cdouble, requires_grad=True)
+        x.real.sum().backward()
+        self.assertEqual(x.grad, torch.ones_like(x))
+
+    # remove this test after gradcheck support is added for non-holomoprphic functions
+    def test_imag(self):
+        x = torch.randn(10, dtype=torch.cdouble, requires_grad=True)
+        x.imag.sum().backward()
+        self.assertEqual(x.grad, -1j * torch.ones_like(x))
+
+    # remove this test after gradcheck support is added for non-holomoprphic functions
+    def test_view_as_real(self):
+        x = torch.randn(10, dtype=torch.cdouble, requires_grad=True)
+        torch.view_as_real(x).sum().backward()
```

Comment:
What about a test like this one EXCEPT:

- you backward through torch.view_as_real AND the complex tensor itself, then validate the gradient on the original complex tensor is correct
- you view_as_real, then view_as_complex from the tensor returned from view_as_real, then view_as_real on that then backward and validate
-

Reply:
Maybe the latter case is sufficiently covered by "as_identity" below?

Reply:
yeah that case is covered by as_identity